### PR TITLE
Arabic Translation

### PR DESCRIPTION
**Pull request by ghost**
A translation from a while back. All the text is translated but it might need some polishing.

### CODE REVIEW

**Comment by ghost** [comment]
There is one other PR for arabic I found from 2018. Might be good to compare. Seems there were some problems with RightToLeft language support but it may be prepared now https://github.com/m2049r/xmrwallet/pull/423

**Comment by ghost** [comment]
If I only knew about this. I checked the translation and it's good aside from spelling "mistakes". One interesting tranlation difference is that of mneumonic seeds. I translated it literally while he translated it as "recovery words". I think my translation is a bit better and I hope the RTL issue is fixed :)

File: app/src/main/res/values-ar/about.xml [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+    <string name="credits_text"><![CDATA[
+        <b>الاعتمادات</b>
+        <br/>
+        m2049r, baltsar777, anhdres, keejef,
+        rehrar, EarlOfEgo, ErCiccione et al.
+        <br/><br/>
+        <a href="https://monerujo.io/">monerujo.io</a>
+        ]]></string>
+
+    <string name="سياسة الخصوصية"><![CDATA[
```
Comment (katant) [inline app/src/main/res/values-ar/about.xml:16]: String name must be `privacy_policy` here. This shoud fix CircleCI build issue

Comment (ghost) [inline app/src/main/res/values-ar/about.xml]: I just changed it. should build now.

File: app/src/main/res/values-ar/strings.xml [hunk 2]
```diff
@@ -0,0 +1,442 @@
...
+    <string name="label_send_txid">معرف المعاملة</string>
+    <string name="label_send_address">عنوان الوجهة</string>
+    <string name="label_send_notes">ملاحظات</string>
+
+    <string name="backup_progress">جار إنشاء النسخة الاحتياطية/string>
+    <string name="archive_progress">جار الأرشفة</string>
+    <string name="rename_progress">جار إعادة التسمية</string>
+    <string name="changepw_progress">جار تغيير كلمة المرور</string>
+
+    <string name="service_progress">جار اختتام الأمور &*8230;\nقد يستغرق هذا برهةً!</string>
```
Comment (katant) [inline app/src/main/res/values-ar/strings.xml:96]: There must be exactly &#8230;

File: app/src/main/res/values-ar/strings.xml [hunk 3]
```diff
@@ -0,0 +1,442 @@
...
+        <item>تلقائي</item>
+        <item>يوم</item>
+        <item>ليل</item>
+    </string-array>
+    <string name="gunther_says">لا شيء هنا\nالرجاء إنشاء أو استعادة محفظة</string>
+
+    <string name="menu_default_nodes">أرجع العقد الافتراضية</string>
+    <string name="toast_default_nodes">الاسترجاع يتقدم بالفعل...</string>
+
+    <string name="node_updated_now">الكتلة الأخيرة: منذ %1&d ثوان</string>
```
Comment (katant) [inline app/src/main/res/values-ar/strings.xml:438]: $d

File: app/src/main/res/values-ar/strings.xml [hunk 4]
```diff
@@ -0,0 +1,442 @@
...
+    <string name="accounts_progress_new">جار إنشاء الحساب</string>
+
+    <string name="toast_ledger_attached">%1$s اربط</string>
+    <string name="toast_ledger_detached">%1$s فصل</string>
+
+    <string name="progress_nfc_write">جار كتابة العلامة</string>
+    <string name="nfc_write_failed">فشلت كتابة العلامة</string>
+    <string name="nfc_write_successful">نجحت كتابة العلامة</string>
+    <string name="nfc_tag_unsupported">العلامة لا تدعم NDEF!</string>
+    <string name="nfc_tag_size">العلامة تزود %!&$لأ بايتاً، لكننا نحتاج %@$ل!</string>
```
Comment (katant) [inline app/src/main/res/values-ar/strings.xml:353]: ```%1$d``` and ```%2$d```

File: app/src/main/res/values-ar/strings.xml [hunk 5]
```diff
@@ -0,0 +1,442 @@
...
+    <string name="status_wallet_connect_failed">فشل الاتصال بالعقدة!\nتأكد من اسم المستخدم/كلمة المرور</string>
+    <string name="status_wallet_connect_wrongversion">نسخة العقدة غير متوافقة - حدث رجاءً!</string>
+    <string name="status_wallet_node_invalid">العقدة غير صحيحة!\nجرب عقدة أخرى.</string>
+    <string name="status_wallet_connect_ioex">لا يمكن الوصول للعقدة!\nحاول مجدداً أو جرب عقدة أخرى.</string>
+
+    <string name="status_wallet_disconnected">منقطعة</string>
+
+    <string name="status_transaction_failed">فشلت المعاملة: $s1%</string>
+
+    <string name="send_xmrto_timeout">!انتظرت طويلاً يا صاحبي!</strig>
```
Comment (katant) [inline app/src/main/res/values-ar/strings.xml:119]: </string>

Comment (katant) [reply app/src/main/res/values-ar/strings.xml:119]: Row 119, not 117

**Review by katant — CHANGES_REQUESTED** [review]
del

File: app/src/main/res/values-ar/strings.xml [hunk 6]
```diff
@@ -0,0 +1,442 @@
...
+    <string name="label_send_btc_xmrto_key_lb">المفتاح السري\nSideshift.ai</string>
+    <string name="label_send_btc_xmrto_key">السري Sideshift.ai مفتاح</string>
+    <string name="label_send_btc_address">عنوان الوجهة BTC</string>
+    <string name="label_send_btc_amount">المبلغ</string>
+
+    <string name="label_send_txid">معرف المعاملة</string>
+    <string name="label_send_address">عنوان الوجهة</string>
+    <string name="label_send_notes">ملاحظات</string>
+
+    <string name="backup_progress">جار إنشاء النسخة الاحتياطية/string>
```
Comment (katant) [inline app/src/main/res/values-ar/strings.xml:91]: ```</string>```

File: app/src/main/res/values-ar/strings.xml [hunk 7]
```diff
@@ -0,0 +1,442 @@
...
+    <string name="label_copy_viewkey">مفتاح الرؤية</string>
+    <string name="label_copy_address">العنوان العام</string>
+    <string name="label_copy_xmrtokey">Sideshift.ai مفتاح</string>
+    <string name="message_copy_viewkey">نسخ مفتاح الرؤية إلى الحافظة!</string>
+    <string name="message_copy_xmrtokey">نسخ مفتاح Sideshift.ai إلى الناسخة!</string>
+    <string name="message_copy_address">نسخ عنوان المحفظة للحافظة!</string>
+    <string name="message_copy_txid">نسخ معرف المعاملة إلى الحافظة!</string>
+    <string name="message_nocopy">النسخ معطل لأسباب أمنية!</string>
+
+    <string name="message_exchange_failed"لا يمكن الحصول على سعر الصرف!\nاستعمل XMR/XMR أو حاول مجدداً</string>
```
Comment (katant) [inline app/src/main/res/values-ar/strings.xml:163]: ```</string>```, RTL symbol?

File: app/src/main/res/values-ar/strings.xml [hunk 8]
```diff
@@ -0,0 +1,442 @@
...
+    <string name="generate_crazypass_label">ملفات المحفظة استرجع كلمة المرور</string>
+
+    <string name="generate_check_key">أدخل مفتاحاً صحيحاً</string>
+    <string name="generate_check_address">أدخل عنواناً صحيحاً</string>
+    <string name="generate_check_mnemonic">أدخل بذرتك ذات 25 كلمة</string>
+
+    <string name="send_notes_hint">الملاحظات الخاصة (اختياري)</string>
+    <string name="send_generate_paymentid_hint">ولد</string>
+    <string name="send_send_label">أنفق مونيرواتي الحبيبة</string>
+    <string name="send_send_timed_label">أرسل مونيرواتي الحبيبة (%1&s)</string>
```
Comment (katant) [inline app/src/main/res/values-ar/strings.xml:215]: ```%1$s```

File: app/src/main/res/values-ar/strings.xml [hunk 9]
```diff
@@ -0,0 +1,442 @@
...
+
+    <string name="details_alert_message">ستظهر بيانات حساسة.\nتفحص خلف ظهرك!</string>
+    <string name="details_alert_yes">أنا في أمان</string>
+    <string name="details_alert_no">أرجعني!</string>
+    <string name="details_title">التفاصيل</string>
+
+    <string name="archive_alert_message">ستحذف المحفظة بعد القيام بنسخة إحتياطية!</string>
+    <string name="archive_alert_yes">نعم، افعل ذلك!</string>
+    <string name="archive_alert_no">لا شكراً!</string>
+
```
Comment (katant) [inline app/src/main/res/values-ar/strings.xml:281]: ```<string-array name="currency" translatable="false">``` before item array

**Comment by katant** [comment]
Some screenshots of app (I don't know arabic, but as I see (and with force RTL + English) there is no issues). But there is still some untranslated strings.
Screenshot_1620635788
Screenshot_1620635870
Screenshot_1620635913
Screenshot_1620635925
Screenshot_1620635936
@phwright should still check app for RTL issues
My compiled testnet APK: https://skyportal.xyz/AACf2tBrS7ItQ7UYpcviWzthPnSqWVUIppYs_OMVh432aA

**Comment by ghost** [comment]
looks good! However, I can't install your APK. I tried it on two different phones, an lg g6 and a samsung galaxy note 4 with a custom rom, and both didn't install it.

**Comment by katant** [comment]
> and both didn't install it.

Its only for x86_64 emulator. 
Universal apk: https://skyportal.xyz/AACkmJMAFFNiLG_V4hIr_-Pz2aD4Dj179TFhZGOoo9rZfQ (but build your own is better)

**Comment by ghost** [comment]
I add the new strings and fixed some typos. should be good now.

**Comment by katant** [comment]
Replace rows 481-484 with
```<string name="tx_details_notes" translatable="false">&lt;span style=\"background-color: #%1$s; color: #%2$s;\"&gt;&nbsp;%3$s&nbsp;&lt;/span&gt;%4$s</string>```
```<string name="receive_subaddress" translatable="false">&lt;span style=\"background-color: #%1$s; color: #%2$s;\"&gt;&nbsp;%3$s&nbsp;&lt;/span&gt;&nbsp;&nbsp;%4$s</string>``` 
Build should pass

**Comment by m2049r** [comment]
cool!
please remove entries with `translatable="false"`.
the build breaks because you have used `&nbsp;` in `strings.xml` but that is an non translatable string anyway. so all will be good when you remove those entries.
thanks!

**Comment by ghost** [comment]
should I remove just the lines with translatable="false" or their contents as well such as
```
<!-- order must be the same as in com.m2049r.xmrwallet.data.Crypto-->
    <string-array name="cryptos" translatable="false">
        <item>Monero</item>
        <item>Bitcoin</item>
        <item>Dash</item>
        <item>Dogecoin</item>
        <item>Ethereum</item>
        <item>Litecoin</item>
    </string-array>
```

**Comment by m2049r** [comment]
with contents - thanks

**Comment by ghost** [comment]
done!

**Pull request merged**